### PR TITLE
Fetch current application from REST on startup & implement missing app fields

### PR DESCRIPTION
**Pull request by Plerx2493**
and dont overwrite CurrentApplication with the partial application coming from ready gw event

# Summary
Fixes an issue with partial application data on the DiscordClient. Not sure if this change was introduce by us or if discord changed application data contained in gw event READY.

# Notes
Also updates DiscordApplication to match discords specs

### CODE REVIEW

File: DSharpPlus/Entities/Application/DiscordApplication.cs [hunk 1]
```diff
@@ -72,23 +72,159 @@ public sealed class DiscordApplication : DiscordMessageApplication, IEquatable<D
...
 
     /// <summary>
     /// Gets the team which owns this application.
     /// </summary>
     public DiscordTeam? Team { get; internal set; }
+    
+    /// <summary>
+    /// Public key used to verify http interactions
+    /// </summary>
+    public string VerifikationKey { get; internal set; }
```
Comment (VelvetToroyashi) [inline DSharpPlus/Entities/Application/DiscordApplication.cs:86]: Typo

File: DSharpPlus/Entities/Application/DiscordApplication.cs [hunk 2]
```diff
@@ -72,23 +72,159 @@ public sealed class DiscordApplication : DiscordMessageApplication, IEquatable<D
...
+
+        this.Bot = transportApplication.Bot is null 
+            ? null 
+            : new DiscordUser(transportApplication.Bot) 
+            {
+                Discord = this.Discord
+            };
+
+        this.GuildId = transportApplication.GuildId;
+        this.Guild = transportApplication.Guild;
```
Comment (VelvetToroyashi) [inline DSharpPlus/Entities/Application/DiscordApplication.cs:184]: `guild.Discord` is not set

**Review by VelvetToroyashi — CHANGES_REQUESTED** [review]

File: DSharpPlus/Entities/Application/DiscordApplication.cs [hunk 3]
```diff
@@ -72,23 +72,164 @@ public sealed class DiscordApplication : DiscordMessageApplication, IEquatable<D
     /// Gets this application's cover image URL.
     /// </summary>
     public override string? CoverImageUrl
-        => $"https://cdn.discordapp.com/app-icons/{this.Id.ToString(CultureInfo.InvariantCulture)}/{this.CoverImageHash}.png?size=1024";
+        =>
+            $"https://cdn.discordapp.com/app-icons/{this.Id.ToString(CultureInfo.InvariantCulture)}/{this.CoverImageHash}.png?size=1024";
```
Comment (VelvetToroyashi) [inline DSharpPlus/Entities/Application/DiscordApplication.cs:76]: Missed this, nit: newline

File: DSharpPlus/Entities/Application/DiscordApplication.cs [hunk 4]
```diff
@@ -78,17 +78,157 @@ public sealed class DiscordApplication : DiscordMessageApplication, IEquatable<D
     /// Gets the team which owns this application.
     /// </summary>
     public DiscordTeam? Team { get; internal set; }
+    
+    /// <summary>
+    /// Public key used to verify http interactions
+    /// </summary>
+    public string VerificationKey { get; internal set; }
```
Comment (VelvetToroyashi) [inline DSharpPlus/Entities/Application/DiscordApplication.cs:85]: nit(consistency): This is VerificationKey, but on the transport object it's VerifyKey (which lines up with the field name)

**Review by VelvetToroyashi — APPROVED** [review]

**Comment by Plerx2493** [comment]
Tested, rtm

**Pull request merged**